src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -266,5 +268,27 @@ public void GetFullPath_CoreTests(string path, string expected)
         {
             Assert.Equal(expected, Path.GetFullPath(path));
         }
+
+        [Theory,
+        InlineData(@"C:\folder\", @"C:\folder"),
+        InlineData(@"C:/folder/", @"C:/folder"),
+        InlineData(@"/folder/", @"/folder"),
+        InlineData(@"\folder\", @"\folder"),
+        InlineData(@"C:\", @"C:\"),
+        InlineData(@"C:/", @"C:/"),
+        InlineData(@"", @""),
+        InlineData(@"/", @"/"),
+        InlineData(@"\", @"\"),
+        InlineData(null, null)]
+        unsafe public void TrimEndingDirectorySeparator_CoreTests(string path, string expected)
+        {
+            string trimmed = Path.TrimEndingDirectorySeparator(path);
+            Assert.Equal(expected, trimmed);
+            Assert.Same(trimmed, Path.TrimEndingDirectorySeparator(trimmed));
+
+            ReadOnlySpan<char> trimmedROS = Path.TrimEndingDirectorySeparator(path.AsSpan());
+            PathAssert.Equal(expected, trimmedROS);
+            Assert.Equal((IntPtr)Unsafe.AsPointer(ref MemoryMarshal.GetReference(trimmedROS)), (IntPtr)Unsafe.AsPointer(ref MemoryMarshal.GetReference(Path.TrimEndingDirectorySeparator(trimmedROS))));
```

Comment:
@stephentoub I found on codebase this code...but maybe there is a better way to do that.

Reply:
For the span case, you can just validate that the trimmed case has the same contents as the trimmed/trimmed case... you don't need to check the refs to make sure they're identical.  I suggested the Same check for the string case since there's an optimization there to return the same string object rather than allocating, but there's no allocation in any case in the span overload.

Reply:
ok